Tasks/Common/nuget-task-common/NuGetXmlHelper.ts

```diff
@@ -0,0 +1,128 @@
+import * as fs from "fs";
+import * as ltx from "ltx";
+
+import * as tl from "vsts-task-lib";
+
+import { INuGetXmlHelper } from "./INuGetXmlHelper";
+
+export class NuGetXmlHelper implements INuGetXmlHelper {
+    constructor(private _nugetConfigPath: string) {
+    }
+
+    public SetApiKeyInNuGetConfig(source: string, apiKey: string): void {
+        tl.debug('Setting ApiKey is not supported without using nuget.exe');
+    }
+
+    public AddSourceToNuGetConfig(name: string, source: string, username?: string, password?: string): void {
+        NuGetXmlHelper._updateXmlFile(this._nugetConfigPath, (xml: any): any => {
+            if (xml) {
+                let xmlPackageSources = NuGetXmlHelper._getLastElement(xml, "packageSources");
```

Comment:
I was surprised at first that _getLastElement will create the element if it is not found.  Either we should change the function name to reflect this, or separate out _getLastElement and _createElement.